feat: add flux source controller

taylorjo02 opened the pull request.

Commit:
feat(source-controller): add flux source-controller

Flux Source Controller v1.5.0 with patched dependencies showing
0 CVE's when scanned with Grype.

Commit:
Merge branch 'main' into add-flux-source-controller

Commit:
fix: build image name

Remove registry path to only leave the name of the image

Commit:
feat: add kustomize-controller

Add kustomize controller and fix help command on both flux images

Commit:
feat: add flux helm controller and patch file

Commit:
feat: add notification controller and cleanup readme

tonybutt commented:
Let's move all this to a single flux package

Something like this dir struct
flux
  default.nix
  *-controller.nix
  
 import the controller nix files into the default.nix
 
 in the flake.nix just import the flux module and call the appropiate function in there and make a patches folder in the flux dir

Commit:
refactor: create flux dir and update import paths

taylorjo02 committed:
fix: Merge branch 'main' into add-flux-source-controller

taylorjo02 committed:
fix: remove .pre-commit-config.yaml

taylorjo02 commented:
Moved it all into the flux dir, and there's the ability to call them individually, or build all 4 together and copy them to docker. also cleaned up linting errors in the readme.

tonybutt commented on flux/default.nix:
```diff
@@ -0,0 +1,43 @@
+  notificationController = import ./notification-controller.nix { inherit pkgs nix2container; };
+  sourceController = import ./source-controller.nix { inherit pkgs nix2container; };
+
+  buildAllControllers = pkgs.writeShellScriptBin "flux-all" ''
```
Swap to pkgs.writeShellApplication
```nix
pkgs.writeShellApplication {
  name = "flux-all"
  text = ''
multiline script here
'';
# or 
text = builtins.readFile ./rel/path/to/my-script.sh
}
```

tonybutt commented on flux/default.nix:
```diff
@@ -0,0 +1,43 @@
+{ pkgs, nix2container, ... }:
+
```
you probably just want to do this

```nix
imports = [
./helm-controller.nix
./kustomize-controller.nix
./notification-controller.nix
];
```

taylorjo02 replied:
I rewrote the import and access the attrset that's returned from each controller build now with this syntax - `helm-controller = (import ./helm-controller.nix { inherit pkgs nix2container; }).helm-controller;`

tonybutt commented on flux/helm-controller.nix:
```diff
@@ -0,0 +1,77 @@
+    };
+  };
+in
+
```
probably just wrap this in an attrset

```nix
{
helm-controller = nix2container...
}
```

tonybutt commented on flake.nix:
```diff
@@ -31,10 +30,21 @@
+        {
+          dragonfly-operator = import ./dragonfly-operator { inherit pkgs nix2container; };
+          external-dns = import ./external-dns { inherit pkgs nix2container; };
+          flux-helm-controller = flux.controllers.helm;
```
once those refactors are there then this becomes flux.helm-controller

taylorjo02 committed:
refactor(flux): simplify imports and update script

Simplify the import of the individual controllers, change to pkgs.write
ShellApplication, and wrap controllers in attrsets
